Spot Illustration: add new Spot Illustration component

troelslenda opened the pull request:
## Which issue does this PR close?

This PR closes #3426

## What is the new behavior?

Introduce spot illustrations.

<!-- Replace this paragraph with a description of the new behaviour after your pull request is merged -->

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this PR contains a breaking change, replace this paragraph with a description of the impact and migration path for existing applications  -->

## Are there any additional context?

### Use of themes in Storybook

There have been added an option to select theme in storybook to illustrate darkmode capabilities of spot-illustrations. There might be are more elegant way to achieve this.

### SVG/Icon adjustments

Technically the icons are, or should be slimmed down to only include one path pr color. These paths should not overlay each other. Theres no reason to paint behind. Furthermore IF enforcing this rule, we could leave out the part called cutaway. This path is merly there to hide the path below, and it dosent work nicely once the illustration is used on other backgrounds.
UX Core has suggested the behaviour seen on the piggy-bank and robot illustration when switching theme, The cutaway should appear "transparent" Why it's entirely removed here. See the video below.

### IconRegistry
Use of IconRegistry was left out of the solution due to coupling issues. This was discussed with @RasmusKjeldgaard earlier.


Having a fallback feature where the other sizes are used in case ie. lg is not jet available for one or more spot illustrations, but keeping the size (24, 40, 56, 95) would lead to alignment issues. You cannot alter the width of strokes, as they are not strokes, but paths with fill. Therefore streaching the illustrations would render the "strokes" with a wrong width.

### Lack of documentation and stories

Until feedback the feature will lack proper documentation and stories. This should not be considered until we are on the right path regarding the implementation..


https://github.com/user-attachments/assets/32141409-9f81-4aec-a4b5-777ab90ad990



<!-- Replace this paragraph with any additional context e.g, explanations, links or screenshots (if any) -->

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

Commit:
cleanup registry.

Commit:
Adding illustrations

Commit:
Add component

Commit:
Update storybook

Commit:
fixing test

Commit:
rename and cleanup icons

Commit:
remove themes again

Commit:
adjust component to use proper sizing and coloring

Commit:
Adding better stories

Commit:
removing excessive styling

Commit:
update icons

Commit:
adressing wrong part for shape

Commit:
adding more examples

Commit:
Add a few more icons

Commit:
Override the primary color

Commit:
adjustments

Commit:
Adding more spotillustrations

Commit:
Cleanup most SVGS.

Commit:
adjusting the rest of the spot illustrations

troelslenda committed:
Merge branch 'develop' into feature/spot-illustrations

RasmusKjeldgaard commented on libs/extensions/angular/.storybook/preview.ts:
```diff
@@ -15,6 +16,9 @@ setCompodocJson(docJson);
     ...defaultParameters,
+    backgrounds: {
+      disable: true,
+    },
```
The backgrounds are useful for the skeleton loader and image banner stories, so consider disabling this on the story level instead.

troelslenda replied:
It was never necessary. It was remains of the temporary introduction of a themes selector.

RasmusKjeldgaard commented on libs/extensions/angular/.storybook/styles.css:
```diff
@@ -1,3 +1,8 @@
 .canvas {
-    background-color: var(--kirby-background-color) !important;
+/*    background-color: var(--kirby-background-color) !important; */
```
Could we do a `.revert-background` class that resets it instead and apply on the relevant story/stories?

troelslenda replied:
☝️

RasmusKjeldgaard commented on libs/extensions/angular/.storybook/styles.css:
```diff
@@ -1,3 +1,8 @@
+body {
+    /* The primary color is overridden in order to show the difference between success and primary. */
+--kirby-primary: LightCoral;
 }
```
This gives the wrong impression, and users will not get what they see when copy pasting, which might confuse.

In the cookbook etc. we accept that primary and success are close for some color schemes, and just leave it as is.

troelslenda replied:
Okay. I See. I Remove this. :)

RasmusKjeldgaard commented on libs/extensions/angular/spot-illustration/src/spot-illustration.component.html:
```diff
@@ -0,0 +1,2 @@
+<ion-icon *ngIf="svg()" [src]="svg()" [class]="size()"></ion-icon>
+<ng-content select="not-found" *ngIf="!svg()"></ng-content>
```
Consider using control flow block syntax now, to match code style. We migrated the whole monorepo over the summer 🌞

troelslenda replied:
Controlflow introduced. 🚀

RasmusKjeldgaard commented on libs/extensions/angular/spot-illustration/src/svgs/robot.none.antenna-dot.body.large.svg:
```diff
@@ -0,0 +1,8 @@
+<svg width="96" height="96" viewBox="0 0 96 96" version="1.1" xmlns="http://www.w3.org/2000/svg"
```
Is the 'NONE' part of the filename of this svg intended?

troelslenda replied:
yes and no. The filename convention presented in the ux/ui guide suggests that every dot seperation has a stricter meaning. leaving out none, would present the body as the highlighted area... The file was named early in the process.

RasmusKjeldgaard replied:
I'll see if I can figure out with the UX'ers what a proper name is

RasmusKjeldgaard commented on libs/extensions/angular/spot-illustration/src/spot-illustration.component.stories.ts:
```diff
@@ -0,0 +1,398 @@
+        <p><em>All requirements and examples are documented in the story's JSDoc comments above.</em></p>
+      </div>
+    `,
+  }),
```
Consider leaving this out of the sidebar with `tags: ['!dev']` on this specific story, instead of creating a custom template.

Then it will only appear on the docs page itself:

```suggestion
  tags: ['!dev']
```

troelslenda replied:
That's a nice take on this. Yeah, you guessed right. It was never needed as a story but just in docs. :)

RasmusKjeldgaard commented on libs/extensions/angular/spot-illustration/src/spot-illustration.component.ts:
```diff
@@ -0,0 +1,36 @@
+
+@Component({
+  selector: 'kirby-x-spot-illustration',
+  standalone: true,
```
Already the default value.
```suggestion
```

troelslenda replied:
fixed.

RasmusKjeldgaard commented on libs/extensions/angular/spot-illustration/src/spot-illustration.component.ts:
```diff
@@ -0,0 +1,36 @@
+@Component({
+  selector: 'kirby-x-spot-illustration',
+  standalone: true,
+  imports: [CardModule, IconModule, CommonModule, IonIcon, IconModule, IconModule],
```
```suggestion
  imports: [IonIcon],
```

troelslenda replied:
fixed.

RasmusKjeldgaard commented on libs/extensions/angular/spot-illustration/src/spot-illustration.component.html:
```diff
@@ -0,0 +1,2 @@
+<ion-icon *ngIf="svg()" [src]="svg()" [class]="size()"></ion-icon>
+<ng-content select="not-found" *ngIf="!svg()"></ng-content>
```
The 'not-found' functionality is quite different from what we do elsewhere in terms of fallback solutions.

To me the only real use-case for fallback logic, other than feedback to the dev during dev-time, is the "oh sh*t the icons are not available runtime because we deployed them to the wrong url/folder"-types of errors. 

So maybe we could bake in a fallback icon instead, that is shown similar to 'cog' in kirby-icon, so the dev experience is the same, and the UI layout is preserved as much as possible for end users?

troelslenda replied:
There was a misunderstanding when approaching this issue at first. I was of the belief that every icon should eventually be represented in every size, but that's not the case. It's more likely to continue like it is now.
So there's no reason for fallbacks and the likes. The initial and undocumented reason for this approach was something like this.

```typescript
  <kirby-x-spot-illustration size="lg" name="piggy-bank">
    <not-found>
      <kirby-x-spot-illustration size="md" name="piggy-bank"></kirby-x-spot-illustration>
    </not-found>
  </kirby-x-spot-illustration>
```

troelslenda replied:
If not, I'd suggest not having any fallaback at all.

RasmusKjeldgaard reviewed (commented):
Stellar work @troelslenda, I am officially blown away.

Here are my comments around the implementation. Once the details have landed I'll add some comments for docs, or if you prefer, clean up the docs myself.

Commit:
Streamline imports, update template syntax, and revert background changes to storybook

Commit:
Remove fallback content and enhance story template with illustration data

troelslenda committed:
Merge branch 'develop' into feature/spot-illustrations

troelslenda committed:
Merge branch 'develop' into feature/spot-illustrations

troelslenda committed:
Merge branch 'develop' into feature/spot-illustrations

Commit:
Refactor spot illustration and update styles for improved clarity
Also modifying the story to better present the component.
Ensuring that the component is properly exported.

troelslenda committed:
Merge branch 'develop' into feature/spot-illustrations

Commit:
Update naming of the svgs.
Added a few
Removed a few

Commit:
Update story

Commit:
Further updating naming to make ux <-> developer communication easier

Commit:
Rename file

Commit:
update naming

troelslenda committed:
Merge branch 'develop' into feature/spot-illustrations

RasmusKjeldgaard reviewed (approved):
✅ ✅ ✅
